File: prime.js
#!/usr/bin/env node --use_strict

function isPrime(num) {
  let num_sqr = Math.sqrt(num)

  if(num_sqr % 1 == 0) return false

  for(let i=2; i<num_sqr; i++) 
    if(num % i == 0) return false

  return true
}

var primes = []
let i = 2;

while(primes.length < parseInt(process.argv[2])) {
  if(isPrime(i)) primes.push(i)
  i++
}

console.log(primes)

File: prime.rb
#!/usr/bin/env ruby

def isPrime(num)
  num_sqr = Math.sqrt(num)

  return false if num_sqr % 1 == 0

  (2..num_sqr.to_i).each do |n|
    return false if num % n == 0
  end

  return true
end

primes = []
i = 2

while primes.count < ARGV.first.to_i do
  primes << i if isPrime(i)
  i += 1
end

puts primes
